### PR TITLE
Allow creation of MSSQL identity columns with user specified seed and…

### PR DESCRIPTION
**Pull request by JRDuncan**
Allow creation of MSSQL identity columns with user specified seed and increment #1576

### CODE REVIEW

File: tests/Phinx/Db/Adapter/SqlServerAdapterTest.php [hunk 1]
```diff
@@ -144,7 +144,22 @@ public function testCreateTableCustomIdColumn()
...
     }
+    public function testCreateTableIdentityColumn()
+    {
+        $table = new \Phinx\Db\Table('ntable', ['id' => false, 'primary_key' => 'id'], $this->adapter);
+        $table->addColumn('id', 'integer', ['identity' => true, 'seed' => 1, 'increment' => 10, ])
+              ->save();
+        $this->assertTrue($this->adapter->hasTable('ntable'));
+        $this->assertTrue($this->adapter->hasColumn('ntable', 'id'));
 
+        $rows = $this->adapter->fetchAll("select cast(seed_value as int) seed_value, cast(increment_value as int) increment_value
```
Comment (garas) [inline tests/Phinx/Db/Adapter/SqlServerAdapterTest.php]: Use uppercase SQL keywords.

Comment (JRDuncan) [reply tests/Phinx/Db/Adapter/SqlServerAdapterTest.php]: Done

File: src/Phinx/Db/Adapter/SqlServerAdapter.php [hunk 2]
```diff
@@ -1186,7 +1186,11 @@ protected function getColumnSqlDefinition(Column $column, $create = true)
         }
 
         if ($column->isIdentity()) {
-            $buffer[] = 'IDENTITY(1, 1)';
+        //$buffer[] = 'IDENTITY(1, 1)';
+            $seed = $column->getSeed() ? $column->getSeed() : 1;
+            $increment = $column->getIncrement() ? $column->getIncrement() : 1;
+         //$buffer[] = 'IDENTITY(' . $column->getSeed() . ',' . $column->getIncrement() . ')';
```
Comment (garas) [inline src/Phinx/Db/Adapter/SqlServerAdapter.php]: - Remove comments
- `x ? x : y` can be simplified to `x ?: y`

File: src/Phinx/Db/Table/Column.php [hunk 3]
```diff
@@ -626,7 +682,10 @@ protected function getValidOptions()
             'properties',
             'values',
             'collation',
-            'encoding',
+        'encoding',
+        /* James Duncan Added seed and increment for MS SQL identity columns */
+        'seed',
+        'increment',
```
Comment (garas) [inline src/Phinx/Db/Table/Column.php]: - Wrong indentation
- Comment unnecessary, there is `git blame` to get this information.

File: tests/Phinx/Db/Adapter/SqlServerAdapterTest.php [hunk 4]
```diff
@@ -144,7 +144,22 @@ public function testCreateTableCustomIdColumn()
         $this->assertTrue($this->adapter->hasColumn('ntable', 'email'));
         $this->assertFalse($this->adapter->hasColumn('ntable', 'address'));
     }
+    public function testCreateTableIdentityColumn()
+    {
+        $table = new \Phinx\Db\Table('ntable', ['id' => false, 'primary_key' => 'id'], $this->adapter);
+        $table->addColumn('id', 'integer', ['identity' => true, 'seed' => 1, 'increment' => 10, ])
```
Comment (garas) [inline tests/Phinx/Db/Adapter/SqlServerAdapterTest.php]: Remove last `,` from array or split array to multiple lines.

File: src/Phinx/Db/Adapter/SqlServerAdapter.php [hunk 5]
```diff
@@ -1186,7 +1186,11 @@ protected function getColumnSqlDefinition(Column $column, $create = true)
...
+            $buffer[] = 'IDENTITY(' . $seed . ',' . $increment . ')';
```
Comment (garas) [inline src/Phinx/Db/Adapter/SqlServerAdapter.php]: Use `sprintf` with `%d` placeholders.

Comment (JRDuncan) [reply src/Phinx/Db/Adapter/SqlServerAdapter.php]: Done

File: src/Phinx/Db/Table/Column.php (hunk 3 above)
Comment (JRDuncan) [reply src/Phinx/Db/Table/Column.php]: Done

File: src/Phinx/Db/Adapter/SqlServerAdapter.php (hunk 2 above)
Comment (JRDuncan) [reply src/Phinx/Db/Adapter/SqlServerAdapter.php]: Done

**Pull request merged**